match.go

```diff
@@ -0,0 +1,72 @@
+package osrm
+
+import geo "github.com/paulmach/go.geo"
+
+// MatchRequest represents a request to the match method
+type MatchRequest struct {
+	Request
```

Comment:
IMHO this `Request` embedding makes for not a really nice API from consumer side - you cannot write like
```go
Match(ctx, MatchRequest{
  Param1: v1,
  Param2: v2,
  ...
})
```
instead you'll have smth like
```go
Match(ctx, MatchRequest{
  Request: Request{
    Param1: v1,
  },
  Param2: v2,
  ...
})
```

Reply:
Agree, lets replace by own fields.

Reply:
updated